chain/chain/src/state_snapshot_actor.rs

```diff
@@ -21,48 +25,70 @@ impl StateSnapshotActor {
 }
 
 impl actix::Actor for StateSnapshotActor {
-    type Context = actix::Context<Self>;
+    type Context = Context<Self>;
 }
 
 #[derive(actix::Message, Debug)]
 #[rtype(result = "()")]
-struct MakeSnapshotRequest {
+struct DeleteSnapshotRequest {
+    /// Optionally send request to create a new snapshot after deleting any existing snapshots.
+    create_snapshot_request: Option<CreateSnapshotRequest>,
+}
+
+#[derive(actix::Message, Debug)]
+#[rtype(result = "()")]
+struct CreateSnapshotRequest {
     /// prev_hash of the last processed block.
     prev_block_hash: CryptoHash,
     /// Shards that need to be present in the snapshot.
     shard_uids: Vec<ShardUId>,
     /// Last block of the prev epoch.
     block: Block,
-    /// Whether to perform compaction.
-    compaction_enabled: bool,
```

Comment:
We don't need `compaction_enabled` param as we can just read this off the snapshot config from trie